### DISCUSSION TITLE
Gameplay transition screen's osu! logo is no longer draggable

### DISCUSSION THREAD

**Discussion by Walavouchey** [ppy/osu#22038] [Q&A] [answered]
it used to be, at least when being able to drag the osu! logo was implemented

https://user-images.githubusercontent.com/36758269/210839396-da0ec7c2-284a-4fdf-a1f1-12f22ccb53df.mp4

2022.1228.0-lazer

**Comment by Joehuu** [comment, marked as answer]
I have regressed this in:
- https://github.com/ppy/osu/pull/20960

The dragging is now handled by a condition:
https://github.com/ppy/osu/blob/ee0bc553565cba5d687257ae8ca3abcbe3ba3db7/osu.Game/Screens/Menu/OsuLogo.cs#L364

`Action` is null in that case, so that's why it's not draggable, clickable, etc.

I personally don't see a point in adding the dragging back, since that screen loads for 1 second at best (without storyboards for now) and you can't really have fun dragging during that time. Not sure what happened to the loading time after that retry, it should be on par with the song select -> player loader or even instant.

**Reply by Walavouchey** [reply]
dragging the logo used to prevent the transition screen from passing (just like interacting with the settings)

**Reply by Joehuu** [reply]
I see. Will try to fix it then.

**Reply by Joehuu** [reply]
I have two solutions for this, but I'm not sure which is better:
- https://github.com/ppy/osu/compare/master...Joehuu:osu:alt-fix-osu-square-drag-area (reverts regressed PR + alternative square fix)
- https://github.com/ppy/osu/compare/master...Joehuu:osu:fix-player-loader-logo-drag (moves `Action` check to all positional methods except drag ones)

**Reply by frenzibyte** [reply]
Either or really. I would pick the second to avoid the idea of proxying things around.

**Reply by Joehuu** [reply]
https://github.com/ppy/osu/pull/22052
